editors/code/src/client.ts

```diff
@@ -95,7 +95,11 @@ export async function createClient(
                     const resp = await next(params, token);
                     if (resp && Array.isArray(resp)) {
                         return resp.map((val) => {
-                            return substituteVSCodeVariables(val);
+                            return prepareVSCodeConfig(val, (key, cfg) => {
+                                if (key === "linkedProjects") {
+                                    cfg[key] = config.discoveredWorkspaces;
```

Comment:
`config.discoveredWorkspaces` is never null/undefiend, but it might be empty. at the moment, this should probably be something like `if (key === linkedProjects && !config.discoveredWorkspaces.isEmpty) { }`, but if you'd like, I can make the definition of `config.discoveredWorkspaces` `JsonProject[] | undefined`?

Reply:
Right an empty check is better here then

Reply:
will make that change, then!

Reply:
alright, should be fixed. sorry about the back-and-forth on this.